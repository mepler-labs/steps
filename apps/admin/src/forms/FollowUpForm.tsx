import React from 'react';
import styled from 'styled-components';
import { Form, Field } from 'react-final-form';
import Panel from 'atoms/Panel';
import { Flex, Box } from 'grid-styled';
import Button from 'atoms/Buttons/Button';
import { remCalc } from 'styles/type';
import { pink, green } from 'styles/colors';

interface Props {
  onSubmit: (data: { weeks: number }) => void;
}

const FollowUpForm: React.SFC<Props> = ({ onSubmit }) => (
  <Form
    onSubmit={onSubmit}
    render={({ handleSubmit }) => (
      <form onSubmit={handleSubmit}>
        <Panel>
          <Flex flexDirection="column" alignItems="center">
            <CalendarIcon className="material-icons">
              calendar_today
            </CalendarIcon>
            <StyledText>
              Let's follow up in
              <Field name="weeks">
                {({ input }) => <NumberInput {...input} />}
              </Field>
              weeks.
            </StyledText>
            <Button>Save</Button>
          </Flex>
        </Panel>
      </form>
    )}
  />
);

const CalendarIcon = styled.i`
  font-size: ${remCalc(50)};
  color: ${pink};
`;

const StyledText = styled.div`
  font-size: ${remCalc(64)};
  margin-bottom: 1rem;
  text-align: center;
`;

const NumberInput = styled.input`
  border: none;
  border-bottom: 2px solid ${green};
  color: ${green};
  font-size: ${remCalc(64)};
  margin: 0 0.5rem;
  outline: none;
  text-align: center;
  width: ${remCalc(50)};
`;

export default FollowUpForm;
