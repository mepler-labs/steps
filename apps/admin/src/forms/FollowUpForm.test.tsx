import { shallow } from 'enzyme';
import FollowUpForm from './FollowUpForm';
import React from 'react';

describe('FollowUpForm.tsx', () => {
  it('renders correctly', () => {
    const onSubmit = jest.fn();
    const wrapper = shallow(<FollowUpForm onSubmit={onSubmit} />);

    expect(wrapper.dive()).toMatchSnapshot();
  });

  it('submits the form', () => {
    const onSubmit = jest.fn();
    const wrapper = shallow(<FollowUpForm onSubmit={onSubmit} />);

    wrapper.simulate('submit', { some: 'data' });

    expect(onSubmit).toHaveBeenCalledWith({ some: 'data' });
  });
});
