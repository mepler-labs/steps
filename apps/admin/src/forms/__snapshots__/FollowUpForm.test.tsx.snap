// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FollowUpForm.tsx renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <Panel>
    <Flex
      alignItems="center"
      flexDirection="column"
      theme={
        Object {
          "breakpoints": Array [
            "40em",
            "52em",
            "64em",
          ],
          "fontSizes": Array [
            12,
            14,
            16,
            20,
            24,
            32,
            48,
            64,
            96,
            128,
          ],
          "space": Array [
            0,
            4,
            8,
            16,
            32,
            64,
            128,
            256,
            512,
          ],
        }
      }
    >
      <styled.i
        className="material-icons"
      >
        calendar_today
      </styled.i>
      <styled.div>
        Let's follow up in
        <Field
          format={[Function]}
          name="weeks"
          parse={[Function]}
        />
        weeks.
      </styled.div>
      <Button>
        Save
      </Button>
    </Flex>
  </Panel>
</form>
`;
